### PR TITLE
Fixing mixed symbol type collection comparisons

### PR DESCRIPTION
**Pull request by JeffAshton**

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/ApiUsage/Logging/LoggingExecutionContextScopeBuilderAnalyzer.cs [hunk 1]
```diff
@@ -184,7 +183,7 @@ CancellationToken ct
 		}
 
 		private static bool IsAwaitable(
-			IImmutableSet<INamedTypeSymbol> taskTypeBuiltins,
+			IImmutableSet<ISymbol> taskTypeBuiltins,
 			INamedTypeSymbol AsyncMethodBuilderAttribute,
 			ITypeSymbol type
 		) {
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/Logging/LoggingExecutionContextScopeBuilderAnalyzer.cs:189]: Down below it was passing `ITypeSymbol` into `.Contains` which causes Linq to take over:

image

File: src/D2L.CodeStyle.Analyzers/ApiUsage/Logging/LoggingExecutionContextScopeBuilderAnalyzer.cs [hunk 2]
```diff
@@ -117,7 +116,7 @@ InvocationExpressionSyntax invocationSyntax
 
 		private static bool IsRunInvocation(
 			SemanticModel model,
-			IImmutableSet<IMethodSymbol> ILoggingExecutionContextScopeBuilderRunSymbols,
+			IImmutableSet<ISymbol> ILoggingExecutionContextScopeBuilderRunSymbols,
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/Logging/LoggingExecutionContextScopeBuilderAnalyzer.cs:119]: image

**Review by omsmith — APPROVED** [review]

File: src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs [hunk 3]
```diff
@@ -66,7 +66,7 @@ internal sealed partial class RpcAnalyzer : DiagnosticAnalyzer {
 
 			RpcTypes rpcTypes = new(
 				RpcAttribute: rpcAttributeType,
-				RpcContexts: ImmutableHashSet.Create<INamedTypeSymbol>(
+				RpcContexts: ImmutableHashSet.Create<ITypeSymbol>(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs:69]: image

File: src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs [hunk 4]
```diff
@@ -48,7 +48,7 @@ internal sealed partial class RpcAnalyzer : DiagnosticAnalyzer {
 				return;
 			}
 
-			ImmutableHashSet<INamedTypeSymbol> knownRpcParameterTypes = ImmutableHashSet.Create<INamedTypeSymbol>(
+			ImmutableHashSet<ITypeSymbol> knownRpcParameterTypes = ImmutableHashSet.Create<ITypeSymbol>(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs:51]: image

File: src/D2L.CodeStyle.Analyzers/Language/StructShouldBeReadonlyAnalyzer.Codefix.cs [hunk 5]
```diff
@@ -54,7 +54,7 @@ CodeFixContext ctx
 
 			private static readonly SyntaxToken READONLY_SYNTAX
 				= SyntaxFactory.Token( SyntaxKind.ReadOnlyKeyword );
-			private async static Task<Document> AddReadonlyModifier(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/Language/StructShouldBeReadonlyAnalyzer.Codefix.cs:57]: There were no awaits.  Not symbol related, just something else I fixed

**Review by j3parker — APPROVED** [review]

**Pull request merged**